diff --git a/android/build.gradle b/android/build.gradle
index 61ed6d6..2f70fd6 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -88,7 +88,8 @@ android {
     externalNativeBuild {
       cmake {
         abiFilters(*reactNativeArchitectures())
-        arguments "-DANDROID_STL=c++_shared",
+        arguments "-DCMAKE_OBJECT_PATH_MAX=1024",
+          "-DANDROID_STL=c++_shared",
           "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
           "-DREACT_NATIVE_DIR=${expoModuleExtension.reactNativeDir}",
           "-DREACT_NATIVE_TARGET_VERSION=${expoModuleExtension.reactNativeVersion.minor}",
